FRFuel.cs

```diff
@@ -576,116 +318,44 @@ public float RandomizeFuelLevel(float fuelLevel)
             float min = fuelLevel / 3f;
             float max = fuelLevel - (fuelLevel / 4);
 
-            return (float)((random.NextDouble() * (max - min)) + min);
-        }
-
-        /// <summary>
-        /// Correctly sets vehicle's fuel level
-        /// </summary>
-        /// <param name="vehicle"></param>
-        /// <param name="fuelLevel"></param>
-        public void VehicleSetFuelLevel(Vehicle vehicle, float fuelLevel)
-        {
-            float max = VehicleMaxFuelLevel(vehicle);
-
-            if (fuelLevel > max)
-            {
-                fuelLevel = max;
-            }
-
-            vehicle.FuelLevel = fuelLevel;
-            vehicle.SetDecor(fuelLevelPropertyName, fuelLevel);
+            return (float)((_random.NextDouble() * (max - min)) + min);
         }
 
         /// <summary>
         /// Returns vehicle's current fuel level
         /// </summary>
         /// <param name="vehicle"></param>
         /// <returns></returns>
-        public float VehicleFuelLevel(Vehicle vehicle)
+        public float GetVehicleFuelLevel(Vehicle vehicle)
         {
             if (vehicle.HasDecor(fuelLevelPropertyName))
             {
                 return vehicle.GetDecor<float>(fuelLevelPropertyName);
             }
             else
             {
-                return 65f;
+                return GetVehicleFuelCapacity(vehicle);
             }
         }
 
         /// <summary>
-        /// Returns vehicle's max fuel level
+        /// Returns vehicle's current fuel level
         /// </summary>
         /// <param name="vehicle"></param>
         /// <returns></returns>
-        public float VehicleMaxFuelLevel(Vehicle vehicle)
+        public float GetVehicleFuelCapacity(Vehicle vehicle)
         {
-            return GetVehicleHandlingFloat(vehicle.Handle, "CHandlingData", "fPetrolTankVolume");
+            return Config.GetInstance().FuelTankCapacityOverride ?? API.GetVehicleHandlingFloat(vehicle.Handle, "CHandlingData", "fPetrolTankVolume");
```

Comment:
This is not the best solution as different vehicles have/should have different tank capacities.

Reply:
Agreed, I saw that in your issues and thought about implementing it, but decided not to overall given all the other changes